File: Program13.cs
/*
	Accept N numbers from user and returns the frequency of even numbers.
*/

using System;

class ArrayX
{
	private int[] iArr;
	
	public ArrayX(int iSize)
	{
		iArr = new int[iSize];
	}
	
	public void accept_display()
	{	
		int iCnt = 0;
		
		Console.WriteLine("Enter array elements : ");
		
		for(iCnt = 0; iCnt < iArr.Length; iCnt++)
		{
			iArr[iCnt] = Convert.ToInt32(Console.ReadLine());
		}
		
		Console.WriteLine("Elements stored in array are : ");
		
		for(iCnt = 0; iCnt < iArr.Length; iCnt++)
		{
			Console.Write(iArr[iCnt]+"\t");
		}
		Console.WriteLine();
	}
	
	public int freqOfEvenElements()
	{
		int iCnt = 0,iCount = 0;
		
		for(iCnt = 0;iCnt < iArr.Length; iCnt++)
		{
			if(((iArr[iCnt])%2) == 0)
			{
				iCount++;
			}	
		}
		
		return iCount;
	}

}

class EntryPointMain
{

	static int Main()
	{
		int iValue = 0;

		Console.WriteLine("Enter number of elements you want to store in array : ");
		iValue = Convert.ToInt32(Console.ReadLine());
		
		ArrayX aObj = new ArrayX(iValue);
		
		aObj.accept_display();
		Console.WriteLine("Count of even elements stored in array is : "+aObj.freqOfEvenElements());			
		
		return 0;
	}
}

File: Program13.c
/*
	Accept N numbers from user and returns the frequency of even numbers.
*/

#include<stdio.h>
#include<stdlib.h>
#include<unistd.h>

int freqOfEvenNumbers(int iSize,int iArr[])
{	
	register int iCnt = 0;
	auto int iCount = 0;
	
	for(iCnt = 0; iCnt < iSize/2; iCnt++)
	{
		if((iArr[iCnt] % 2) == 0)
		{
			iCount++;
		}
	}
	return iCount;
}

int main(void)
{
	int freqOfEvenNumbers(int,int *);
	
	auto int iValue = 0,iRet = 0,*iPtr = NULL;	

	register int iCnt = 0;

	// Accepting size of array from user	
	printf("Enter number of elements you want to store in array : \n");
	scanf("%d",&iValue);
	
	// Dynamic memory allocation for array
	iPtr = (int *)malloc(iValue * sizeof(int));
	if(iPtr == NULL)
	{
		perror("malloc : ");
		_exit(EXIT_FAILURE);
	}
	
	// Accepting the elements
	printf("\nEnter elements : \n");
	for(iCnt = 0;iCnt < iValue; iCnt++)
	{
		scanf("%d",&iPtr[iCnt]);
	}
	
	// Printing the elements
	printf("\nElements stored in array are : \n");
	for(iCnt = 0;iCnt < iValue; iCnt++)
	{
		printf("%d\t",iPtr[iCnt]);
	}	
	printf("\n");
	
	// Function call	
	iRet = freqOfEvenNumbers(iValue,iPtr);
	printf("\nCount of even numbers present in array are : %d\n",iRet);
	
	// Freed the dynamic memory allocated	
	if(iPtr)
	{
		free(iPtr);
	}	
	
	exit(EXIT_SUCCESS);
		
}
